lib/puppet/provider/puppet_agent_end_run/puppet_agent_end_run.rb

```diff
@@ -32,8 +32,13 @@ def needs_upgrade?
     return false if desired_version == 'present'
 
     if desired_version == 'latest'
-      latest_version = @resource.catalog.resource('package', 'puppet-agent').parameters[:ensure].latest
-      desired_version = latest_version.match(%r{^(?:[0-9]:)?(\d+\.\d+(\.\d+)?(?:\.\d+))?}).captures.first
+      # Package name might be different to puppet-agent, hence we need to look it up.
+      package_name = @resource.catalog.resource('class', 'puppet_agent')[:package_name]
```

Comment:
If `package_name` can be nil (not sure that's possible?), then it'd be good to set a default:

```suggestion
      package_name = @resource.catalog.resource('class', 'puppet_agent')[:package_name] || 'puppet-agent'
```

Reply:
I don't think it can (realistically) be nil - but having a sane default still seems like a  good idea.